### ISSUE TITLE
Multi-NIC - Add new template parameter to specify NIC index of EC2 instance

### ISSUE THREAD

**Issue by megamattzilla** [F5Networks/f5-aws-cloudformation#132]
## Multi-NIC - Add new template parameter to specify NIC index of EC2 instance
The current standalone 3-NIC ARM template allows you to deploy an F5 VE with 3 interfaces in this EC2 index order: 
0= MGMT interface
1= External Interface 
2= Internal Interface

Some AWS components such as instance based NLB target groups can only use the index 0 IP address for related functionality. This causes problems using instance based selection because the MGMT interface may not be the preferred interface target and instead you may want to target a different interface such as the External interface. 

Using non-instance based (interface) targeting has its own caveats that don't exist when using instance based targeting. One example is NLB using specific NIC targeting must perform source address translation whereas instance based targeting does not require source address translation. 

Currently, there is no discernable way in the CFT template to change the NIC index order to the following:
0= External Interface 
1= Internal Interface
2= MGMT Interface 

Ideally, I would like to see a parameterized option in the Arm template to change the index order. In the CFT template, there is a **DeviceIndex** attribute where this is specified:
```
"NetworkInterfaces": [
     {
      "Description": "Management Interface",
      "DeviceIndex": "0",
      "NetworkInterfaceId": {
       "Fn::If": [
        "useDynamicManagementIpAddr",
        {
         "Ref": "Bigip1ManagementInterface"
        },
        {
         "Ref": "Bigip1StaticManagementInterface"
        }
       ]
      }
     },
     {
      "Description": "Public or External Interface",
      "DeviceIndex": "1",
      "NetworkInterfaceId": {
       "Fn::If": [
        "useDynamicExternalIpAddr",
        {
         "Ref": "Bigip1subnet1Az1Interface"
        },
        {
         "Ref": "Bigip1Staticsubnet1Az1Interface"
        }
       ]
      }
     },
     {
      "Description": "Private or Internal Interface",
      "DeviceIndex": "2",
      "NetworkInterfaceId": {
       "Fn::If": [
        "useDynamicInternalIpAddr",
        {
         "Ref": "Bigip1InternalInterface"
        },
        {
         "Ref": "Bigip1StaticInternalInterface"
        }
       ]
      }
     }
    ],
```

However even if you change the index number at this level, it seems that the cloud libraries still setup the Big-IP interface configuration in the original order of MGMT, EXT, INT. 

## Template
https://github.com/F5Networks/f5-aws-cloudformation/tree/master/supported/standalone/3nic/existing-stack/byol

## Severity Level

Severity: 3

**Comment by shyawnkarim** [comment]
Simply changing the DeviceIndex attribute does not work because this feature would also require additional code changes, a BIG-IP reboot, and then finally the nic swap.

I've added this enhancement to our backlog and we are tracking it internally with ID ESECLDTPLT-2299.

**Comment by shyawnkarim** [comment]
Closing due to age. These legacy templates are now in maintenance mode and are being replaced by our next-generation templates available in the [Cloud Templates 2.0 GitHub repo](https://github.com/F5Networks/f5-aws-cloudformation-v2).
